scripts/deployment/l2/deploy_04_gnosis_safe_same_address_multisig.js

```diff
@@ -0,0 +1,56 @@
+/*global process*/
+
+const { ethers } = require("hardhat");
+const { LedgerSigner } = require("@anders-t/ethers-ledger");
+
+async function main() {
+    const fs = require("fs");
+    const globalsFile = "globals.json";
+    const dataFromJSON = fs.readFileSync(globalsFile, "utf8");
+    let parsedData = JSON.parse(dataFromJSON);
+    const useLedger = parsedData.useLedger;
+    const derivationPath = parsedData.derivationPath;
+    const providerName = parsedData.providerName;
+    let EOA;
+
+    const provider = await ethers.providers.getDefaultProvider(providerName);
+    const signers = await ethers.getSigners();
+
+    if (useLedger) {
+        EOA = new LedgerSigner(provider, derivationPath);
+    } else {
+        EOA = signers[0];
+    }
+    // EOA address
+    const deployer = await EOA.getAddress();
+    console.log("EOA is:", deployer);
+
+    // Transaction signing and execution
+    console.log("4. EOA to deploy GnosisSafeMultisig");
+    const GnosisSafeMultisig = await ethers.getContractFactory("GnosisSafeSameAddressMultisig");
+    console.log("You are signing the following transaction: GnosisSafeSameAddressMultisig.connect(EOA).deploy()");
+    const gnosisSafeMultisig = await GnosisSafeMultisig.connect(EOA).deploy();
+    const result = await gnosisSafeMultisig.deployed();
+
+    // Transaction details
+    console.log("Contract deployment: GnosisSafeMultisig");
+    console.log("Contract address:", gnosisSafeMultisig.address);
+    console.log("Transaction:", result.deployTransaction.hash);
+
+    // Writing updated parameters back to the JSON file
+    parsedData.gnosisSafeSameAddressMultisigImplementationAddress = gnosisSafeMultisig.address;
```

Comment:
```suggestion
    parsedData.gnosisSafeSameAddressMultisigImplementationAddress = gnosisSafeSameAddressMultisig.address;
```

Reply:
why doing all this when we can reuse old name variables?